### ISSUE TITLE
Expose `resolveModuleNames`

### ISSUE THREAD

**Issue by arcanis** [TypeStrong/ts-loader#861]
Hi!

Would it be possible to consider exposing the `resolveModuleNames` function (set in `servicesHost.ts`) through a configuration settings?

I'm currently experimenting integrating Yarn PnP with TypeScript through ts-loader, and my initial results are very promising. I think I could get something working just by going through this hook, which would unlock pretty interesting options!

**Comment by johnnyreilly** [comment]
Go for it @arcanis! Experiment on!

**Comment by arcanis** [comment]
Will aim to set up a PR in the next week or so 👍

**Comment by johnnyreilly** [comment]
Cool - look forward to it. If tests cause you problems (many people bump on this) then stick up the PR and we can collaborate

**Comment by johnnyreilly** [comment]
> It's slow because in the current implementation, names are resolved twice, and the results are not cached.

By the way, did you know about this option recently added to ts-loader which adds caching? I'm planning to turn it on by default (currently off) as I've had no reports of issues:

https://github.com/TypeStrong/ts-loader/blob/master/README.md#experimentalfilecaching-boolean-defaultfalse

**Comment by thatsmydoing** [comment]
> By the way, did you know about this option recently added to ts-loader which adds caching? I'm planning to turn it on by default (currently off) as I've had no reports of issues:

Yeah, I've tried it but it didn't make much of an improvement for us. The caching happens too late in the process it seems. Module resolution seems to be a recursive operation and that seems to be what's taking up the time.

I took a look at the current PR and the changes are also happening at too low a level for the change I was thinking of making. I'll experiment a bit more with how the resolution could be sped up.
